drivers/backup/portworx/portworx.go

```diff
@@ -1092,6 +1093,120 @@ func (p *portworx) GetBackupUID(ctx context.Context, orgID, backupName string) (
+	var rulesInfo api.RulesInfo
+	var uid string
+	for i := 0; i < len(appList); i++ {
+		if prePostFlag == "pre" {
+			CreatePreRule = true
+			if _, ok := AppParameters[appList[i]]["pre_action_list"]; ok {
+				podSelector = append(podSelector, ps[appList[i]])
+				actionValue = append(actionValue, AppParameters[appList[i]]["pre_action_list"])
+				backgroundVal, _ := strconv.ParseBool(AppParameters[appList[i]]["background"])
+				background = append(background, backgroundVal)
+				podVal, _ := strconv.ParseBool(AppParameters[appList[i]]["runInSinglePod"])
+				runInSinglePod = append(runInSinglePod, podVal)
+				// Here user has to set env for each app container if required in the format container-<app name>
+				// This is for the optional parameter 'Container' for creating rules
+				containerName := fmt.Sprintf("%s-%s", "container", appList[i])
+				container = append(container, os.Getenv(containerName))
+			} else {
+				log.Infof("Pre rule not required for this application")
+			}
+		} else {
+			CreatePostRule = true
+			if _, ok := AppParameters[appList[i]]["post_action_list"]; ok {
+				podSelector = append(podSelector, ps[appList[i]])
+				actionValue = append(actionValue, AppParameters[appList[i]]["post_action_list"])
+				backgroundVal, _ := strconv.ParseBool(AppParameters[appList[i]]["background"])
+				background = append(background, backgroundVal)
+				podVal, _ := strconv.ParseBool(AppParameters[appList[i]]["runInSinglePod"])
+				runInSinglePod = append(runInSinglePod, podVal)
+				// Here user has to set env for each app container if required in the format container-<app name>
+				// This is for the optional parameter 'Container' for creating rules
+				containerName := fmt.Sprintf("%s-%s", "container", appList[i])
+				container = append(container, os.Getenv(containerName))
+			} else {
+				log.Infof("Post rule not required for this application")
+			}
+		}
+	}
+	totalRules := len(actionValue)
+	if totalRules == 0 {
+		log.Info("Rules not required for the apps")
+		return true, ""
+	}
+
+	rulesInfoRuleItem := make([]api.RulesInfo_RuleItem, totalRules)
+	for i := 0; i < totalRules; i++ {
+		ruleAction := api.RulesInfo_Action{Background: background[i], RunInSinglePod: runInSinglePod[i],
+			Value: actionValue[i]}
+		var actions = []*api.RulesInfo_Action{&ruleAction}
+		rulesInfoRuleItem[i].PodSelector = podSelector[i]
+		rulesInfoRuleItem[i].Actions = actions
+		rulesInfoRuleItem[i].Container = container[i]
+		rulesInfo.Rules = append(rulesInfo.Rules, &rulesInfoRuleItem[i])
+	}
+	RuleCreateReq := &api.RuleCreateRequest{
+		CreateMetadata: &api.CreateMetadata{
+			Name:  ruleName,
+			OrgId: orgID,
+		},
+		RulesInfo: &rulesInfo,
+	}
+	ctx, err := backup.GetAdminCtxFromSecret()
+	if err != nil {
+		log.Errorf("Failed to fetch px-central-admin ctx: [%v]", err)
+		return false, ""
+	}
+
+	_, err = p.CreateRule(ctx, RuleCreateReq)
+	if err != nil {
+		log.Errorf("Failed to create backup rules: [%v]", err)
```

Comment:
retrun the error and fail the test with it

Reply:
done!